src/main/scala/cosas/ops/types/SerializeDenotations.scala

```diff
@@ -0,0 +1,44 @@
+package ohnosequences.cosas.ops.types
+
+import ohnosequences.cosas._, types._, typeSets._, fns._
+
+// TODO msg not found
+trait SerializeDenotations[Denotations <: AnyTypeSet, V]
+extends Fn2[Denotations, Map[String,V]] with
```

Comment:
Point by point:

- you do get a map with elements of that typeset/record; look at the code
``` scala
def serializeTo[V](implicit
  serialize: RT#PropertySet#Raw SerializeDenotations V
): Either[SerializeDenotationsError, Map[String,V]] = serialize(entryRaw, Map())
```
- you can do that check on the typesets/records, but about what the serializers do. This way what you do will fail (as it should) instead of arbitrarily keeping the values of the second map. `++` on `Map` is wrong, and should only be used _after_ checking for keys being there.